src/Common/src/System/Runtime/InteropServices/StringBuffer.cs

```diff
@@ -42,6 +42,20 @@ public unsafe StringBuffer(string initialContents)
         }
 
         /// <summary>
+        /// Instantiate the buffer with a copy of the specified string.
+        /// </summary>
+        public unsafe StringBuffer(StringBuffer initialContents)
+            : base(0)
+        {
+            // We don't pass the count of bytes to the base constructor, appending will
+            // initialize to the correct size for the specified initial contents.
+            if (initialContents != null)
+            {
+                Append(initialContents);
+            }
```

Comment:
Since we're not using initialContents to size the new StringBuffer at all all, it looks like this doesn't really add any value over the caller just doing:
```C#
var sb = new StringBuffer();
sb.Append(initialContents);
```
It also looks like this ctor is only used in one place in the code you added.  Do we need the additional ctor?